### PR TITLE
handle loading, error, no jobs, success scenarios

### PR DESCRIPTION
**Pull request by raymond-sde**

### CODE REVIEW

**Comment by chaynsaw** [comment]
@raymond-sde you have a bug here that I noticed doesn't occur on main. 
When you search for jobs and you try to swipe right on a card, you'll get this error on expo web:

image

File: components/SearchForJobs.tsx [hunk 1]
```diff
@@ -10,55 +10,93 @@ import axios from "axios";
...
+      case ViewState.LOADING:
+        return <p>Loading...</p>;
+      case ViewState.ERROR:
+        return renderBack("Something went wrong.");
+      case ViewState.NO_DATA:
+        return renderBack("No jobs found. Blame Covid.");
+      case ViewState.SUCCESS:
+        return (
+          <View>
+            {data.length && (
```
Comment (chaynsaw) [inline components/SearchForJobs.tsx:68]: Don't understand what's going on here... so if data.length is not falsy (0), then it renders? I would have thought this statement evaluates to a boolean.

Comment (raymond-sde) [reply components/SearchForJobs.tsx:68]: If data.length is truthy, it renders the right block, else it renders nothing

Comment (raymond-sde) [reply components/SearchForJobs.tsx:68]: its a common short circuit technique. if the left side of && is truthy...execute right side. if left side of && is falsy...right side doesn't execute.

**Pull request merged**